server/src/main/java/org/opensearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -629,6 +654,13 @@ public ShardIterator preferNodeActiveInitializingShardsIt(Set<String> nodeIds) {
         return new PlainShardIterator(shardId, preferred);
     }
 
+    public boolean isNodeWeighedAway(WeightedRouting weightedRouting, DiscoveryNodes nodes, String localNodeId, double defaultWeight) {
+        String attVal = nodes.get(localNodeId).getAttributes().get(weightedRouting.attributeName());
+        // If weight for a zone is not defined, considering it as 1 by default
+        Double weight = weightedRouting.weights().getOrDefault(attVal, defaultWeight);
+        return weight.intValue() == WeightedRoutingMetadata.WEIGHED_AWAY_WEIGHT;
+    }
+
```

Comment:
we are not using this . So are including nodes with weight 0 as well ?

Reply:
Will remove this. Not needed